dashboard/app/controllers/api/v1/pd/workshops_controller.rb

```diff
@@ -117,7 +117,10 @@ def show
   def update
     adjust_facilitators
 
-    unless current_user.permission?(UserPermission::WORKSHOP_ORGANIZER) || current_user.permission?(UserPermission::PROGRAM_MANAGER) || current_user.permission?(UserPermission::WORKSHOP_ADMIN)
+    unless current_user.permission?(UserPermission::WORKSHOP_ORGANIZER) ||
+      current_user.permission?(UserPermission::PROGRAM_MANAGER) ||
+      current_user.permission?(UserPermission::WORKSHOP_ADMIN) ||
+      Pd::CourseFacilitator.exists?(facilitator: current_user, course: Pd::Workshop::COURSE_CSF)
```

Comment:
I wonder if we should define this as a CanCan abilty, then do something like
`unless @current_ability.can? :update_regional_partner, @workshop` in line 120?

Instead of deleting from `params`, which is destructive, this would be cleaner if we moved the logic into the strong parameter function `workshop_params`, only permitting `:regional_partner_id` if those conditions are met ^